recipe/meta.yaml

```
@@ -1,69 +1,115 @@
 {% set name = "numcodecs" %}
-{% set version = "0.14.1" %}
+{% set version = "0.16.3" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 00a364924fd2d600bcce6e2ced96b47c40eb5f9d84bf4b0207aa208d9ce6cd1c
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 53d705865faaf0a7927c973af3777532001c8fbb653de119c1e844608614d799
   patches:
-    - patches/0001-unvendor-c-blosc.patch
+    - patches/0001-Unvendor-deps.patch
 
 build:
   number: 0
-  # numcodecs is not currently supported on s390x
-  skip: True  # [py<311 or s390x]
+  skip: true  # [py<311]
   script:
     - export DISABLE_NUMCODECS_AVX2=""  # [unix]
     - export DISABLE_NUMCODECS_SSE2=""  # [osx and arm64]
     - set DISABLE_NUMCODECS_AVX2=""  # [win]
     - {{ PYTHON }} -m pip install --no-deps . -vv --no-build-isolation
+  entry_points:
+    - numcodecs.blosc = numcodecs.zarr3:Blosc
+    - numcodecs.lz4 = numcodecs.zarr3:LZ4
+    - numcodecs.zstd = numcodecs.zarr3:Zstd
+    - numcodecs.zlib = numcodecs.zarr3:Zlib
+    - numcodecs.gzip = numcodecs.zarr3:GZip
+    - numcodecs.bz2 = numcodecs.zarr3:BZ2
+    - numcodecs.lzma = numcodecs.zarr3:LZMA
+    - numcodecs.shuffle = numcodecs.zarr3:Shuffle
+    - numcodecs.delta = numcodecs.zarr3:Delta
+    - numcodecs.bitround = numcodecs.zarr3:BitRound
+    - numcodecs.fixedscaleoffset = numcodecs.zarr3:FixedScaleOffset
+    - numcodecs.quantize = numcodecs.zarr3:Quantize
+    - numcodecs.packbits = numcodecs.zarr3:PackBits
+    - numcodecs.astype = numcodecs.zarr3:AsType
+    - numcodecs.crc32 = numcodecs.zarr3:CRC32
+    - numcodecs.crc32c = numcodecs.zarr3:CRC32C
+    - numcodecs.adler32 = numcodecs.zarr3:Adler32
+    - numcodecs.fletcher32 = numcodecs.zarr3:Fletcher32
+    - numcodecs.jenkins_lookup3 = numcodecs.zarr3:JenkinsLookup3
+    - numcodecs.pcodec = numcodecs.zarr3:PCodec
+    - numcodecs.zfpy = numcodecs.zarr3:ZFPY
 
 requirements:
   build:
+    - {{ stdlib('c') }}
     - {{ compiler('c') }}
-    - patch # [unix]
-    - m2-patch # [win]
   host:
     - python
     - pip
     - cython
-    - setuptools >=64
-    - wheel
-    - toml  # [py<311]
+    - toml
     - py-cpuinfo
-    - blosc 1.21.3
-    - lz4-c 1.9.4
-    - zstd {{ zstd }}
+    - setuptools >=77
     - setuptools-scm >=6.2
-    - numpy 2
+    - numpy >2
```

```suggestion
    - numpy {{ numpy }}
```
This yields an optimal run range.

---

This solution wasn't compatible with python 3.13 so I restricted pinning to it. I also put a comment explaining why.